force-app/main/default/classes/WeatherService.cls

```diff
@@ -0,0 +1,33 @@
+global class WeatherService {
+    @AuraEnabled(cacheable=true)
+    public static WeatherData getCurrentWeather(String location){ 
+           
+        WeatherAPIKey__mdt weaherApiKey = [SELECT Key__c FROM WeatherAPIKey__mdt LIMIT 1];  
+        String apiKey = weaherApiKey.Key__c;
+
+        HttpRequest req = new HttpRequest();
+        req.setEndpoint('callout:WeatherAPI?q='+ location + '&units=metric' + '&appid=' + apiKey);
+        req.setMethod('GET');
+        Http http = new Http();
+        HttpResponse resp = http.send(req);
+       
+        Map<String, Object> weatherMap = (Map<String, Object>) JSON.deserializeUntyped(resp.getBody());
```

Comment:
don't use Hungarian notation

Reply:
one more thing is it needs to check if response is success first and after that work with body of it